### ISSUE TITLE
Key Maestro/Bar of Repository ID, rather than name/url

### ISSUE THREAD

**Issue by ChadNedzlek** [dotnet/arcade#11564]
I've seen several renaming of repositories in FR lately, and this causes a lot of problems with our engineering systems, because we use the full repository URL as the "key" in a lot of places, which changes when the repo is renamed (an otherwise non-destructive operation that _shouldn't_ break everything).

It shouldn't be that hard for us to use the repository ID's, rather than their display names, as the keys in our systems without too much work, I would hope which could really prevent a lot of dangerous operations (like having to log into the production database and mess with installation keys).

**Comment by ChadNedzlek** [comment]
We could have it as something like "github:12345" (where 12345 is either the "id" or "nodeId" field from GitHub), and "azo:GUID" when it's the GUID in AzDO.  We could have a display table that we only use when rendering things for "last time I looked this up, what was it called" (since it's harder to go the other direction... from stable ID to name), but this would solve a lot of very dangerous problems going forward.

**Comment by michellemcdaniel** [comment]
I'd like to talk about this in triage. I think it's a good idea, but it's probably something that would need to go to a ProdCon epic. @tkapin @premun @mmitche for visibility.

**Comment by ChadNedzlek** [comment]
No, the installation ID requires, to my knowledge, directly manipulating the database, or uninstalling the app from the org and reinstalling it (it's not installed a repo level, so you've gotta do the whole org).  I don't see anywhere in the code where we'd "update" the installation information except in those cases. In fact, you _can't_ create a new subscription, because we can't figure out how to talk to GitHub, because we think we aren't installed.

https://github.com/dotnet/arcade/issues/11516 can help with the installation stuff (we just shouldn't store it, there's very little benefit to doing so and is the messiest part of all this).

I would agree that because of all the tecg-debt in arcade-services, a change like this is likely very expensive, and potentially not worth doing... but it IS causing a bunch of pain for users and FR, so it's at least worth talking about.

**Comment by premun** [comment]
> it's probably something that would need to go to a ProdCon epic

@michellemcdaniel I don't think there's a "ProdCon epic" really but we have to figure out a place for work like this soon. The only epic we have right now is the arcade-services debt and the VMR-lite one (which doesn't really have anything to do with this).

Let's discuss this tomorrow on Epic Review?

**Comment by michellemcdaniel** [comment]
Sound good, @premun. /cc @Chrisboh, also. I'm happy to just not pay attention to these kinds of issues in triage, but if we can agree on some sort of home for them, that would be, imo, better.

**Comment by tkapin** [comment]
@michellemcdaniel - Agree. The exact issue / project management process for the product construction is yet to be designed as it depends on various things that are not decided yet (such as whether we'll continue to track these issues in arcade or rather in a dedicated repo). This process will be determined as part of #11628 which I assume will start sometimes at the beginning of the new calendar year. 

In the meantime, I agree we need something to keep the triage process running and an epic would be the seamless solution. I've logged #11629 to be used as container for the product construction issues, feel free to start assigning these issues to that epic.

**Comment by michellemcdaniel** [comment]
Thanks so much!

**Comment by tkapin** [comment]
@ChadNedzlek, @ilyas1974 - The idea of using the repo ID (combined with a repo origin identification to allow for various providers) sound very reasonable to me as the repo ID's purpose is exactly to provide the for the repo. However from reading the discussion above, it seems that despite it needs attention on FR from time-to-time, the effort to implement this is significant and also introduces non-negligible risk of breaking things. To help me reasoning about the ROI, could you estimate how much effort (devhours, devdays) were spent on resolving this by FR? Thanks
